pkg/logging/logging.go

```diff
@@ -133,6 +138,36 @@ func LoadLogConfig(dataStore, ns, id string) (LogConfig, error) {
 	return logConfig, nil
 }
 
+func loggingProcessAdapter(driver Driver, dataStore string, config *logging.Config) error {
+	if err := driver.PreProcess(dataStore, config); err != nil {
+		return err
+	}
+	var wg sync.WaitGroup
+	wg.Add(3)
+	stdout := make(chan string, 10000)
+	stderr := make(chan string, 10000)
+	processLogFunc := func(reader io.Reader, dataChan chan string) {
+		defer wg.Done()
+		defer close(dataChan)
+		scanner := bufio.NewScanner(reader)
+		for scanner.Scan() {
+			if scanner.Err() != nil {
+				return
+			}
+			dataChan <- scanner.Text()
+		}
+	}
+
+	go processLogFunc(config.Stdout, stdout)
+	go processLogFunc(config.Stderr, stderr)
+	go func() {
+		defer wg.Done()
+		driver.Process(stdout, stderr)
+	}()
+	wg.Wait()
```

Comment:
`PreProcess` , `Process` and `PostProcess` should be executed sequentially

Reply:
I think it's already executed sequentially here